stream-legal-entity/legal-entity-core/src/main/java/com/backbase/stream/LegalEntitySaga.java

```diff
@@ -159,9 +143,113 @@ public Mono<LegalEntityTask> executeTask(@SpanTag(value = "streamTask") LegalEnt
+                streamTask.info(LEGAL_ENTITY, PROCESS_CONTACTS, FAILED, legalEntity.getExternalId(), legalEntity.getInternalId(),
+                        "Legal Entity: %s does not have any Users", legalEntity.getExternalId());
+                return Mono.just(streamTask);
+            }
+        }
+        return contactsSaga.executeTask(createContactsTask(streamTask.getId(), legalEntity.getExternalId(), serviceAgreement.getExternalId(), externalUserId, AccessContextScope.SA, serviceAgreement.getContacts()))
+                .flatMap(contactsTask -> requireNonNull(Mono.just(streamTask)))
+                .then(Mono.just(streamTask));
+    }
+
+    private String getParticipantUser(ServiceAgreement serviceAgreement) {
+        if (!isEmpty(serviceAgreement.getParticipants())) {
+            Optional<LegalEntityParticipant> participants = serviceAgreement.getParticipants()
+                    .stream()
+                    .filter(LegalEntityParticipant::getSharingUsers)
+                    .findFirst();
+            if (participants.isPresent()) {
+                LegalEntityParticipant participant = participants.get();
+                if (!isEmpty(participant.getUsers())) {
+                    return getOptionalUserId(participant.getUsers().stream().findFirst());
+                } else if (!isEmpty(participant.getAdmins())) {
+                    return getOptionalUserId(participant.getAdmins().stream().findFirst());
+                }
+            }
+        }
+        return null;
+    }
+
+    private String getOptionalUserId(Optional<String> optionalUserId) {
+        return optionalUserId.isPresent() ? optionalUserId.get() : null;
+    }
+
+    private ContactsTask createContactsTask(String streamTaskId, String externalLegalEntityId, String externalServiceAgreementId, String externalUserId, AccessContextScope scope, List<ExternalContact> contacts) {
+        var contactData = new ContactsBulkPostRequestBody();
+        contactData.setIngestMode(IngestMode.UPSERT);
+        contactData.setAccessContext(createExternalAccessContext(externalLegalEntityId, externalServiceAgreementId, externalUserId, scope));
+        contactData.setContacts(externalContactMapper.toMapList(contacts));
+        return new ContactsTask(streamTaskId + "-" + "contacts-task", contactData);
+    }
+
+    private ExternalAccessContext createExternalAccessContext(String externalLegalEntityId, String externalServiceAgreementId, String externalUserId, AccessContextScope scope) {
+        ExternalAccessContext accessContext = new ExternalAccessContext();
+        accessContext.setExternalLegalEntityId(externalLegalEntityId);
+        accessContext.setExternalServiceAgreementId(externalServiceAgreementId);
+        accessContext.setExternalUserId(externalUserId);
+        accessContext.setScope(scope);
+        return accessContext;
+    }
+
+    private String getUserExternalId(List<JobProfileUser> users) {
+        if (users == null) {
+            return null;
```

Comment:
the main reasons why we have the Optional wrapper is to avoid checking for nulls in your business log, please use `Optional` in your checks

Reply:
Fixed, changed to Optional